### PR TITLE
[4.3] Updating additional npm dependencies

### PR DESCRIPTION
**Pull request by Hackwar**
After creating #38605, I also did update all other npm dependencies with a `npm update`. This update also includes the changes from #38605.

### CODE REVIEW

**Comment by Hackwar** [comment]
npm fails with an error in the CSS.

**Comment by brianteeman** [comment]
and the error is?

**Comment by Hackwar** [comment]
```
Error: Undefined variable.

   ╷
20 │   @each $color, $value in $theme-colors-rgb {
   │                           ^^^^^^^^^^^^^^^^^
   ╵
  build/media_source/templates/administrator/atum/scss/_root.scss 20:27            @import
  build/media_source/templates/administrator/atum/scss/vendor/_bootstrap.scss 8:9  @import
  build/media_source/templates/administrator/atum/scss/template.scss 8:9           @import
  build/media_source/templates/administrator/atum/scss/template-rtl.scss 1:9       root stylesheet

npm ERR! code 1
npm ERR! path /********/src
npm ERR! command failed
```

**Comment by brianteeman** [comment]
yeah sorry - I saw it after I posted and you beat me to it before I could say whoops

**Comment by brianteeman** [comment]
Pretty sure I know what it is - just testing locally first.

**Comment by brianteeman** [comment]
The problem is that npm update has updated to bootstrap 5.2 which definitely cannot be merged at this time.

in the package.json
change
     "bootstrap": "^5.1.3",
to
    "bootstrap": "~5.1.3",

and then rerun npm update

**Comment by Hackwar** [comment]
Fixed.

**Comment by brianteeman** [comment]
Sorry this still fails - you can not blindly run npm update as there are several build scripts that run based on them.

In this specific case codemirror has been updated so the script that generated the codemirror plugin has to be run and will result in at least an updated codemirror.xml file.

Sadly despite their being a pull request for this specific update it was ignored and hence you have the problem here. https://github.com/joomla/joomla-cms/pull/38343

In other words if you want to do an npm update and/or a composer update then you must also run the build scripts before you submit the pull request.

I've mentioned this a few times and that this should be automated but sadly as ever it fell on deaf ears.

From memory this effects jooa11y, codemirror, tinymce, skipto, accessibility and maybe some more

**Comment by Hackwar** [comment]
No, I don't know about these tasks and you never specifically pointed that information towards me. Now I know about this and I will write it on my long list of todos. I try to not be deaf to stuff like this. That is why I listened to Dimitris regarding Renovatebot or why I'm still working on https://github.com/joomla/framework.joomla.org/pull/46, besides lots of other stuff. And no, I'm not happy that all of this is taking so long.

**Comment by brianteeman** [comment]
Apologies if you were not one of the people I have mentioned this to. I definitely said it George, Harold and Roland and Dmitris already knew because he wrote the code.

**Comment by Hackwar** [comment]
I'm rather fixing renovatebot to get this right also for the future instead of merging this one. Closing.
